src/assets/src/components/edit.tsx

```diff
@@ -93,8 +96,118 @@ function HostEditor(props: HostEditorProps) {
+            <div className="input-group-append">
+                <MeetingTypeDropdown options={options} defaultValue="default" onChangeValue={props.onChangeDropdownState}/>
+            </div>
+            <div className="input-group-append">
+                <button className="btn btn-success" type="submit" disabled={props.disabled}>
+                    + Add Attendee
+                </button>
+            </div>
+        </form>
+    );
+}
+
+interface AllowedMeetingTypesFormProps {
+    queue: QueueHost;
+    backends: {[backend_type: string]: string};
+    onSubmit: (allowedTypes: string[]) => void;
+    disabled: boolean;
+}
+
+function GetCheckboxes(labels: string[], checkedLabels: string[]) {
+    const checkboxes = [];
+    for (var i = 0; i < labels.length; i++) {
+        checkboxes.push({checked: checkedLabels.includes(labels[i])});
+    }
+    return checkboxes;
+}
+
+function AllowedMeetingTypesForm(props: AllowedMeetingTypesFormProps) {
+    const [checkedCheckboxes, setCheckedCheckboxes] = useState(GetCheckboxes(Object.keys(props.backends), props.queue.allowed_backends));
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        let checked = checkedCheckboxes;
+        console.log(checked[Object.keys(props.backends).indexOf(event.target.name)].checked);
+        checked[Object.keys(props.backends).indexOf(event.target.name)].checked = !checked[Object.keys(props.backends).indexOf(event.target.name)].checked;
+        console.log(checked[Object.keys(props.backends).indexOf(event.target.name)].checked);
+        setCheckedCheckboxes(checked);
+    }
+    const submit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        const currentCheckboxes = checkedCheckboxes;
+        if (checkedCheckboxes.length === 0) {
+            setCheckedCheckboxes(GetCheckboxes(Object.keys(props.backends), props.queue.allowed_backends));
+        }
+        let choice = [];
+        for (var i = 0; i < currentCheckboxes.length; i++) {
+            if (currentCheckboxes[i].checked) {
+                choice.push(Object.keys(props.backends)[i]);
+            }
+        }
+        props.onSubmit(choice);
+    }
+    const checkbox = (label: string) => {
+        return (
+        <>
+        <label htmlFor={label}>{props.backends[label]}:</label>
+        <input name={label} type="checkbox" checked={checkedCheckboxes[Object.keys(props.backends).indexOf(label)].checked} onChange={handleChange}/>
+        </>);
+    }
+    const checkboxes = [];
+    for (var i = 0; i < Object.keys(props.backends).length; i++) {
+        var thisType = Object.keys(props.backends)[i];
+        if (i === 0) {
+            checkboxes.push(checkbox(thisType));
+        } else {
+            let thisBox = <>
+                <div className="input-group-append">
+                    {checkbox(thisType)}
+                </div> 
+            </>
+            checkboxes.push(thisBox);
+        }
+    }
```

Comment:
This for block could be simplified quite a bit:
```
const checkboxes = Object.keys(props.backends).map((b, i) =>
    i === 0 ? checkbox(b) : <div className="input-group-append">{checkbox(b)}</div>
);
```
Using map/filter is generally better than a for loop when you're only transforming or filtering a set of values, because it makes your intent crystal clear and limits the scope of mistakes that could be made within the block.

What is the `i === 0` check for, though? Seeing if it's some default value?